### PR TITLE
Add aggregation list to node info

### LINKED ISSUE #52057: Add aggregation list to node info
We would like to develop a test to make sure that transform covers newly added aggregations. For this purpose we need to have a list of all available aggregations in both core and x-pack. The most reliable way to do it would be to add the list of all registered aggregations (possibly grouped by their type) to node info similar to how we provide the list of available processors.

### PR DESCRIPTION
**Pull request by imotov**
There are at least two use cases (transform tests and telemetry mapping
updated) in which we need to have an access to full list of supported
aggregations. This PR adds this list to the node info API.

Fixes #52057

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        for (Map.Entry<String, Set<String>> e : aggs.entrySet()) {
+            out.writeString(e.getKey());
+            out.writeVInt(e.getValue().size());
+            for (String type : e.getValue()) {
+                out.writeString(type);
+            }
+        }
+    }
+
+    public Map<String, Set<String>> getProcessors() {
```
Comment (jakelandis) [inline server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java:71]: getProcessors() ? I assume the name is off, but is this even needed ?

File: rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      nodes.info: {}
+  - set:
+      nodes._arbitrary_key_: node_id
+
+  - do:
+      nodes.info:
+        metric: [ aggregations ]
+
+  # if this test failed because a new aggregation was added, please open an issues in the elastic/telemetry repository
+  # so they can update the mapping accordingly
```
Comment (jakelandis) [inline rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml:19]: this comment seems abit out of place, and probably shouldn't reference a private repo. 

also, I guess I don't understand the output.  is it aggregrations.??.types = ?? . Is there anything that prevents checking for known types

**Review by jakelandis — COMMENTED** [review]
changes look good.  a couple comments. do you have an example of the output ?

File: rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml (hunk 2 above)
Comment (imotov) [reply rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml:19]: I was sure I removed it. This comment is out of place. Good catch.

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java (hunk 1 above)
Comment (imotov) [reply server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java:71]: Copy and paste error. Will fix.

File: rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml (hunk 2 above)
Comment (imotov) [reply rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/40_aggs.yml:19]: The output looks like this:
```
      "aggregations" : {
        "adjacency_matrix" : {
          "types" : [
            "other"
          ]
        },
        "auto_date_histogram" : {
          "types" : [
            "boolean",
            "date",
            "numeric"
          ]
        },
        "avg" : {
          "types" : [
            "boolean",
            "date",
            "histogram",
            "numeric"
          ]
        },
        "boxplot" : {
          "types" : [
            "histogram",
            "numeric"
          ]
        },
        "cardinality" : {
          "types" : [
            "boolean",
            "bytes",
            "date",
            "geopoint",
            "geoshape",
            "ip",
            "numeric",
            "range"
          ]
        },
        ....
```

and both types and aggs change depending on license that this is ran under.

**Review by jakelandis — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            out.writeString(e.getKey());
+            out.writeVInt(e.getValue().size());
+            for (String type : e.getValue()) {
+                out.writeString(type);
+            }
+        }
+    }
+
+    public Map<String, Set<String>> getAggregations() {
+        return aggs;
```
Comment (not-napoleon) [inline server/src/main/java/org/elasticsearch/search/aggregations/AggregationInfo.java:72]: We're leaking a mutable reference here.  It doesn't look like we mutate it anywhere, might be worth wrapping it to be immutable, if the object creation cost for the wrapper isn't too much.

**Review by not-napoleon — APPROVED** [review]
I think this is fine to merge, but if we can make `AggregationInfo` immutable, I think that would be better.

**Pull request merged**